### PR TITLE
[RHTAPBUGS-441]add generatecomponentname option

### PR DESCRIPTION
**Pull request by yangcao77**
Part of https://issues.redhat.com/browse/RHTAPBUGS-441

add new optional spec field in CDQ call `generateComponentName`. if the value is set, the name generation will skip the check for component existence and always have random characters appended.

### CODE REVIEW

File: api/v1alpha1/componentdetectionquery_types.go [hunk 1]
```diff
@@ -35,6 +35,12 @@ type ComponentDetectionQuerySpec struct {
 	// Optional.
 	// +optional
 	Secret string `json:"secret,omitempty"`
+
+	// If should generate the component name instead of a predicted default value
```
Comment (kim-tsao) [inline api/v1alpha1/componentdetectionquery_types.go:39]: Do you mean "It should generate"?

**Review by kim-tsao — APPROVED** [review]
/lgtm but we should update the description before merging

File: api/v1alpha1/componentdetectionquery_types.go (hunk 1 above)
Comment (yangcao77) [reply api/v1alpha1/componentdetectionquery_types.go:39]: I meant "if". as it's a flag and defines if the component name should generate random chars

Comment (yangcao77) [reply api/v1alpha1/componentdetectionquery_types.go]: updated the description to make it more clear

**Review by johnmcollier — APPROVED** [review]

**Review by kim-tsao — APPROVED** [review]

**Pull request merged**